Sources/MongoSwift/BSON/DocumentIterator.swift

```diff
@@ -127,23 +127,45 @@ public class DocumentIterator: IteratorProtocol {
             return [:]
         }
 
-        // skip the values preceding startIndex. this is more performant than calling next, because
-        // it doesn't pull the unneeded key/values out of the iterator
-        for _ in 0..<startIndex { _ = iter.advance() }
-
-        var output = Document()
+        var excludedKeys: [String] = []
 
-        // TODO: SWIFT-224: use va_list variant of bson_copy_to_excluding to improve performance
-        for _ in startIndex..<endIndex {
+        for _ in 0..<startIndex {
             if let next = iter.next() {
-                output[next.key] = next.value
+                excludedKeys.append(next.key)
             } else {
                 // we ran out of values
                 break
             }
         }
 
-        return output
+        // skip the values between startIndex and endIndex. this is more performant than calling next, because
+        // it doesn't pull the unneeded key/values out of the iterator
+        for _ in startIndex..<endIndex {
+            if !iter.advance() {
+                // we ran out of values
+                break
+            }
+        }
+
+        while let next = iter.next() {
+            excludedKeys.append(next.key)
+        }
+
+        switch excludedKeys.count {
+        case 0:
+            return [:]
+        default:
+            let firstExclude = excludedKeys.removeFirst()
+            guard let bson = bson_new() else {
+                fatalError("Invalid BSON data")
+            }
+            do {
+                try self.withVaListOfCStrings(excludedKeys, doc._bson, bson, firstExclude)
+            } catch {
+                fatalError("Failed to copy strings")
```

Comment:
nit: rather than duplicating the error message here, which could end up getting out of sync, we could do something like `fatalError("Error creating document subsequence: \(error)")` which will provide a little more context about the call site and will insert the string description of the error that was caught.

Reply:
Done.